src/swap/SwapScreen.tsx

```diff
@@ -114,6 +113,46 @@ export function SwapScreenSection({ showDrawerTopNav }: { showDrawerTopNav: bool
     [swapAmount]
   )
 
+  const calculatePriceImpactData = () => {
+    if (!exchangeRate) {
+      return { isPriceImpactMissing: false, priceImpactExceedsThreshold: false }
+    }
+
+    const priceImpact = exchangeRate.estimatedPriceImpact
+      ? new BigNumber(exchangeRate.estimatedPriceImpact).dividedBy(100)
+      : null
+
+    const isPriceImpactMissing = priceImpact === null
+    const priceImpactExceedsThreshold =
+      !!priceImpact && priceImpact.gte(priceImpactWarningThreshold)
+
+    return { isPriceImpactMissing, priceImpactExceedsThreshold, priceImpact }
+  }
+
+  const { isPriceImpactMissing, priceImpactExceedsThreshold } = useMemo(() => {
+    return calculatePriceImpactData()
+  }, [exchangeRate])
```

Comment:
ah, this is more complex than i imagined but i think you did this to save doing the priceImpactExceedsThreshold logic multiple times right? this is okay, what do you think about changing the type of exchangeRate.estimatedPriceImpact to be already a big number and divided by 100 or null when it reaches this component? so then your useEffect hook can dispatch analytics when `!exchangeRate.estimatedPriceImpact || exchangeRate.estimatedPriceImpact.gte(priceImpactWarningThreshold)` and your render can use `!exchangeRate.estimatedPriceImpact` and `exchangeRate.estimatedPriceImpact.gte(priceImpactWarningThreshold)` directly? it would save a little bit of overhead to not have the extra variables that you need to keep synchronised?

Reply:
I did these changes, I think the code looks better. Thanks!